### ISSUE TITLE
css-modules break styles.css file

### ISSUE THREAD

**Issue by sillyslux** [gatsbyjs/gatsby#798]
It seems that `gatsby build` writes to /pubic/styles.css when it processes `build-css` [(code)](https://github.com/gatsbyjs/gatsby/blob/8e337d17b5a61846e15b326bb56c50c88cc5e7d2/lib/utils/webpack.config.js#L133) and then it overwrites /public/styles.css during `build-html` [(code)](https://github.com/gatsbyjs/gatsby/blob/8e337d17b5a61846e15b326bb56c50c88cc5e7d2/lib/utils/webpack.config.js#L138).
Only the first version is good, it contains all global css and all css-modules code, but the 2nd write, after `build-html`, includes only styles from css-modules.
I've observed this behavior with both, my project and [gatsby-starter-default](https://github.com/gatsbyjs/gatsby-starter-default).
Right now i'd be left without 99% of css, so i changed `build-html` to write to styles.modules.css for now.
Before i started to use css-modules i had no such problems, so i guess it's related to css-modules.

**Comment by KyleAMathews** [comment]
Oh strange... no I'm not sure why we did things this way or why it hasn't been reported as a bug before. The build-javascript step definitely needs the extract text plugin to remove css strings from the created JavaScript bundle but build-html doesn't...

Could you test this? Try removing the extract text plugin altogether from `build-html` and to be safe, change the output name for `build-javascript` to something like `.styles.css` to just hide it in a dot file?

**Comment by sillyslux** [comment]
So, i have used `noProductionJavascript=true` until now and didn't even notice that `build-javascript` also writes to `style.css`. I've updated gatsby to 0.12.43 and changed file names to `styles.css.css`, `styles.html.css` and `styles.js.css`.
And now it gets a little complicated:
first off [example.module.css](https://github.com/gatsbyjs/gatsby-starter-default/blob/master/pages/example.module.css) isn't really regular css syntax. Is this allowed? I still don't know much about css-modules, but it renders just fine in styles.css.css. Only the other two files, both come out like this:
```
:root {
    --color: pink
}
._12mMbi9pYZjHnytkkBkl9v {
    & ul {
        margin: 0;
        padding: 0;
        list-style: none
    }
    & li {
        display: inline-block
    }
...
```
which looks more like `less`, so i've renamed `example.module.css` to `example.module.less` and the two broken files look a little better:
```
:root {
    --color: pink
}
._3djxVUozqi9dpSabpsM29p ul {
    margin: 0;
    padding: 0;
    list-style: none
}
._3djxVUozqi9dpSabpsM29p li {
    display: inline-block
}
...
```
but that's still only the code from example.module.css, all the other css is missing in these two files.

As i've said, i only use styles.css.css (i.e. styles.modules.css) anyway, so i'm good for now (and until some colleagues start cloning...)

Before removing extract-text-plugin from `build-html` **and** `build-javascript` bundle-size was 506475Bytes, after it was 506349Bytes, so even a little smaller, if there is some gatsby-starter-default css-code in this bundle.js, that i do not know. Here is the [commit](https://github.com/sillyslux/gatsby/commit/b0a79e6be5176fd3e4cec69ec7a64188e0dce965) i've made.

**Comment by KyleAMathews** [comment]
The `example.module.css` is using future css syntax powered by http://cssnext.io/

I'm pretty sure I understand what the problem is here. I'll make a quick PR to fix this and make a new release.

**Comment by sillyslux** [comment]
Ty for the quick fix!
Then i can delete my local workarounds :)
Ah, and you were able to reproduce? Then it is strange indeed that no-one complained until now...

**Comment by KyleAMathews** [comment]
Yeah... perhaps people only use either CSS modules or regular CSS so they didn't notice that they were interfering with each other.

Just released 0.12.44 — please upgrade and test and report back if things are working now!

**Comment by sillyslux** [comment]
Earlier it wrote to styles.modules.css (my workaround), now it writes to .styles.css, same thing and good enough for me. Thanks again.
